File: AtCoder_Beginners_Selection/atcorder_shift_only.js
function Main(input) {
  const text_array = input.split("\n");
  let text_array_number = text_array[1].split(" ").map(x => +x);
  // console.log(text_array_number);

  let count = 0;
  while (!text_array_number.map(x => +x).map(x => x % 2).includes(1)) {
    text_array_number = text_array_number.map(x => x / 2);
    // console.log(text_array_number);
    count++;
  };
  console.log(count);

};

Main(require('fs').readFileSync('/dev/stdin', 'utf8'));

File: AtCoder_Beginners_Selection/atcorder_shift_only.py
n = int(input())
a_arr = list(map(int, input().split(" ")))

count = 0
flag = False

while True:
  for i in range(n):
    if a_arr[i] % 2 == 0:
      a_arr[i] //= 2
    else:
      print(count)
      flag = True
      break
  count += 1
  if flag:
    break
